core-processor/src/executor.rs

```diff
@@ -351,9 +351,11 @@ where
     };
     let (termination, memory, ext) = match execute() {
         Ok(report) => {
-            let (mut termination, mut memory, ext) = report
-                .into_parts()
-                .map_err(SystemExecutionError::SyscallFunc)?;
+            let BackendReport {
+                termination_reason: mut termination,
+                memory_wrap: mut memory,
+                ext,
+            } = report;
```

Comment:
Impl fn, use here and below
```suggestion
            (mut termination, mut memory, ext) = report.into_parts();
```

Reply:
Mmm, don't see reason to do that. I've not touched BackendReport in this PR. Also I think into_parts is less informative when read code and should be used when it's really have to